source/slang/slang-ir-string-hash.cpp

```diff
@@ -0,0 +1,108 @@
+// slang-ir-string-hash.cpp
+#include "slang-ir-string-hash.h"
+
+#include "slang-ir.h"
+#include "slang-ir-insts.h"
+
+namespace Slang {
+
+static void _findGetStringHashRec(IRInst* inst, List<IRGetStringHash*>& outInsts)
+{
+    for (IRInst* child = inst->getFirstDecorationOrChild(); child; child = child->getNextInst())
+    {
+        if (IRGetStringHash* getInst = as<IRGetStringHash>(child))
+        {
+            outInsts.add(getInst);
+        }
+        _findGetStringHashRec(child, outInsts);
+    }
+}
+
+void replaceGetStringHash(IRModule* module, SharedIRBuilder& sharedBuilder, StringSlicePool& ioPool)
+{
+    IRBuilder builder;
+    builder.sharedBuilder = &sharedBuilder;
+
+    builder.setInsertInto(module->getModuleInst());
+
+    List<IRGetStringHash*> insts;
+    _findGetStringHashRec(module->getModuleInst(), insts);
+
+    // Then we want to add the GlobalHashedString instruction in the root
+    for (auto inst : insts)
+    {
+        IRStringLit* stringLit = inst->getStringLit();
+        ioPool.add(stringLit->getStringSlice());
+        
+        // Okay work out what the hash is
+        const int hash = GetHashCode(stringLit->getStringSlice());
```

Comment:
Using the existing `GetHashCode` used within the Slang implementation is probably fine for a first pass, but I think we should switch to something that is documented and reproducible/testable eventually, both to ensure that we don't accidentally change the hash values seen by applications as a result of trying to optimize our own code, and also so that a client who wants to could compute their own hashes by using the same algorithm we have documented ourselves as using.